src/main/java/duke/TaskList.java

```diff
@@ -43,26 +44,20 @@ void parseEventFromFile(String[] tokens) {
     }
 
     /**
-     * Method to handle adding a task to the underlying list, and trigger corresponding <code>Ui</code> event.
+     * Adds a task to the list.
      * @param task Target <code>Task</code> to be added to list.
-     * @param ui Instance of <code>Ui</code> associated with the calling instance of <code>Duke</code>.
      */
-    void addTaskToList(Task task, Ui ui) {
+    void addTaskToList(Task task) {
         this.tasks.add(task);
-        ui.displayMessage("Got it. I've added this task:\n" +
-                task.toString() +
-                "\nNow you have " +
-                tasks.size() +
-                " tasks in the list\n");
     }
 
     /**
-     * Method to check the validity of the format of a <code>ToDo</code> task, and adds it to the list.
+     * Checks the validity of the format of a <code>ToDo</code> task, and adds it to the list.
      * @param tokens <code>String[]</code> of arguments from <code>Parser</code>.
-     * @param ui Instance of <code>Ui</code> associated with the calling instance of <code>Duke</code>.
      * @throws DukeException In the event that the name of the task is not specified.
+     * @return The added <code>ToDo</code>.
      */
-    void addToDo(String[] tokens, Ui ui) throws DukeException {
+    ToDo addToDo(String[] tokens) throws DukeException {
         StringBuilder sb = new StringBuilder();
```

Comment:
Perhaps you could instantiate one StringBuilder instance class wide, then calling sb.setLength(0) to clear the sb at the start of a method, so that you don't have to create multiple sb instances.